contracts/feature-tests/composability/forwarder/src/call_async.rs

```diff
@@ -106,6 +106,62 @@ pub trait ForwarderAsyncCallModule {
         #[indexed] payment: &BigUint,
     );
 
+    #[endpoint]
+    fn forward_async_retrieve_multi_transfer_funds(
+        &self,
+        to: ManagedAddress,
+        with_callback: bool,
+        #[var_args] token_payments: ManagedVarArgs<MultiArg3<TokenIdentifier, u64, BigUint>>,
+    ) {
+        let mut arg_buffer = ManagedArgBuffer::new_empty(self.raw_vm_api());
+        let mut nr_transfers = 0usize;
+
+        for multi_arg in token_payments.into_iter() {
+            let (token_identifier, token_nonce, amount) = multi_arg.into_tuple();
+
+            arg_buffer.push_arg(token_identifier);
+            arg_buffer.push_arg(token_nonce);
+            arg_buffer.push_arg(amount);
+
+            nr_transfers += 1;
+        }
+
+        if with_callback {
+            let mut callback_data_serializer = elrond_wasm::HexCallDataSerializer::new(
+                &b"retrive_funds_multi_transfer_callback"[..],
+            );
+            callback_data_serializer.push_argument_bytes(&nr_transfers.to_be_bytes()[..]);
+
+            self.raw_vm_api().storage_store_tx_hash_key(
+                &self
+                    .types()
+                    .managed_buffer_from(callback_data_serializer.as_slice()),
+            );
+        }
+
+        self.raw_vm_api().async_call_raw(
+            &to,
+            &self.types().big_uint_zero(),
+            &self
+                .types()
+                .managed_buffer_from(&b"retrieve_multi_funds_async"[..]),
+            &arg_buffer,
+        );
+    }
+
+    #[callback]
+    fn retrive_funds_multi_transfer_callback(&self, expected_nr_transfers: usize) -> SCResult<()> {
```

Comment:
I think the raw callback in forwarder-raw already posts everything it receives in storage/logs. Maybe merge with that.